cli/src/pcluster/resources/compute_node/user_data.sh

```diff
@@ -151,6 +151,18 @@ write_files:
         done;
         export HOME="${!HOME_BAK}"
       }
+
+      function publish_startup_time
+      {
+        StartTime=$1
+        EndTime=$2
+        METRIC_NAMESPACE="ParallelCluster"
+        METRIC_NAME="StartupTime"
+        INSTANCE_ID=$(cat /etc/parallelcluster/slurm_plugin/slurm_node_spec.json | jq -r .instance_id)
```

Comment:
This cannot work if the scheduler is not slurm.

Reply:
Updated code so we only put the metric if the scheduler is slurm